src/components/DateInput/Day.tsx

```diff
@@ -1,35 +1,105 @@
 import React from 'react';
 import Box, { BoxProps } from '../Box';
+import styled from '@emotion/styled';
 import dayjs, { Dayjs } from 'dayjs';
-import { noop } from '../../utils/helpers';
+import { noop, addOpacity } from '../../utils/helpers';
 
 export interface DayProps {
   day?: number;
+  isLastRow?: boolean;
   month: number;
   year: number;
   daySelected?: Dayjs;
+  dayRangeSelected?: [Dayjs?, Dayjs?];
   onDaySelect?: (date: Dayjs) => void;
 }
 
-const Cell: React.FC<BoxProps> = props => (
+const PseudoOuterCell: React.FC<BoxProps> = props => (
   <Box
     display="table-cell"
-    padding={2}
+    verticalAlign="middle"
     textAlign="center"
-    borderRadius="circle"
     cursor="pointer"
-    verticalAlign="middle"
-    _selected={{ backgroundColor: 'blue-400' }}
-    _hover={{ backgroundColor: 'blue-300' }}
+    position="relative"
     {...props}
   />
 );
 
-const MemoCell = React.memo(Cell);
+const OuterCell = styled(PseudoOuterCell)`
+  [aria-relevant]:hover {
+    background-color: ${({ theme }) => theme.colors['blue-300']};
+  }
+  [aria-placeholder] {
+    position: absolute;
+    z-index: 1;
+    width: 50%;
+    height: 100%;
+    top: 0;
+  }
+
+  &[aria-busy='true'] + [aria-selected='true'],
+  &[aria-selected='true'] + [aria-busy='true'],
+  &[aria-selected='true'] + &[aria-selected='true'] {
+    [aria-placeholder] {
+      background-color: ${({ theme }) => addOpacity(theme.colors['navyblue-100'], 0.2)};
+    }
+  }
+
+  &[aria-selected='true'] + &[aria-selected='true'] [aria-placeholder] {
```

Comment:
Lots of selectors in order to show the placeholder as a range along with border-radius for each row.